### PR TITLE
Implement proper palette attributes for the scatter plot palette

### PR DESCRIPTION
**Pull request by couet**
Implement properly the scatter plot palette attributes as requested here: https://github.com/root-project/root/issues/15922
Simplify scatter.C

### CODE REVIEW

File: hist/hist/inc/TScatter.h [hunk 1]
```diff
@@ -57,6 +57,8 @@ class TScatter : public TNamed, public TAttLine, public TAttFill, public TAttMar
    Double_t  GetMinMarkerSize() const {return fMinMarkerSize;} ///< Get the smallest marker size used to paint the markers
    TGraph   *GetGraph()  const {return fGraph;}                ///< Get the graph holding X and Y positions
    TH2F     *GetHistogram() const;                             ///< Get the graph histogram used for drawing axis
+   TAxis    *GetXaxis() const ;
```
Comment (ferdymercury) [inline hist/hist/inc/TScatter.h:60]: Why are these two functions needed?

Comment (ferdymercury) [reply hist/hist/inc/TScatter.h:60]: If it's for ease of use, then they should be three?  (GetZaxis  in addition) ?

**Review by ferdymercury — APPROVED** [review]
This looks great, thanks a lot.

I attached a couple of questions/comments.

This solves the second point of #15922

The first one (controlling the z axis range) not yet.

File: hist/hist/inc/TScatter.h (hunk 1 above)
Comment (couet) [reply hist/hist/inc/TScatter.h:60]: Yes , ease of use, GetZaxis is coming. The PR is not finished yet. I am looking at distancetoprimitive now, One cannot pick the axis....

**Pull request merged**